tests/client/test_lock.py

```diff
@@ -133,6 +136,23 @@ def get_lock(self, redis, *args, **kwargs):
         with pytest.raises(LockError):
             await lock.extend(10)
 
+    @pytest.mark.asyncio()
+    async def test_concurrent_lock_acquire(self, r):
+        lock = self.get_lock(r, 'test', timeout=1)
+
+        async def coro(lock):
+            is_error_raised = False
+            await lock.acquire(blocking=True)
+            await asyncio.sleep(1.5)
+            try:
+                await lock.release()
+            except LockError as exc:
+                is_error_raised = True
+            return is_error_raised
+
+        results = await asyncio.gather(coro(lock), coro(lock))
+        assert not (results[0] and results[1])
```

Comment:
This assertion fails when I run this test locally. I believe the assertion is incorrect.

Both tasks/coroutines sleep for longer than the lifetime of the lock. As a consequence, both will encounter a LockError when attempting to release the lock. Hence, the correct assertion would be `assert (result[0] and result[1])`

Reply:
@ei

> This assertion fails when I run this test locally. I believe the assertion is incorrect.
> 
> Both tasks/coroutines sleep for longer than the lifetime of the lock. As a consequence, both will encounter a LockError when attempting to release the lock. Hence, the correct assertion would be `assert (result[0] and result[1])`

I think you are right, the test result is random which is not appropriate